osler/inventory/tests/tests.py

```diff
@@ -30,61 +32,61 @@ def test_drugupdate_url(self):
         path = reverse('inventory:drug-update', kwargs={'pk':1})
         self.assertEqual(resolve(path).view_name, 'inventory:drug-update')
 
-def drug_dict():
-
-    drug_dict = {
-        'name': 'Somedrug',
-        'unit': MeasuringUnit.objects.create(name='someunit'),
-        'dose': 1000.0,
-        'stock': 10,
-        'expiration_date': '2040-01-01',
-        'lot_number': 'ABCDEFGH',
-        'category': DrugCategory.objects.create(name='somecategory'),
-        'manufacturer': Manufacturer.objects.create(name='somemanufacturer')
-    }
-
-    return drug_dict
-
 class TestModels(TestCase):
 
     def test_drugcategory_creation(self):
-        category = DrugCategory.objects.create(name='somecategory')
+        assert DrugCategory.objects.count() == 0
+        category = factories.DrugCategoryFactory()
         assert isinstance(category, DrugCategory)
-        self.assertEqual(category.__str__(), category.name)
+        assert DrugCategory.objects.count() == 1
 
     def test_measuringunit_creation(self):
-        measuringunit = MeasuringUnit.objects.create(name='someunit')
+        assert MeasuringUnit.objects.count() == 0
+        measuringunit = factories.MeasuringUnitFactory()
         assert isinstance(measuringunit, MeasuringUnit)
-        self.assertEqual(measuringunit.__str__(), measuringunit.name)
+        assert MeasuringUnit.objects.count() == 1
```

Comment:
I don't actually thing you need to test your factories like this. Normally, your "model tests" would test functions on your models themselves, like `pre_dispense` or `can_dispense`

Reply:
Do you think it's redundant to have unit tests for each model? I looked online to learn about testing, and they suggested that I do unit tests on everything before doing more integrated (i.e. creating drugs via forms) tests. I liked the unit tests since if something goes wrong, I know exactly what is wrong, but let me know if you just want the functional/integrated tests. I'm fine either way.

Reply:
I think it's ok not to have it - we don't really do it anywhere else in Osler and it's not a huge hassle to pinpoint the issue in the more integrated tests unless they get super long